cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -165,11 +165,26 @@ spOutsFromCardanoTx cardanoTx = forM (Pl.getCardanoTxOutRefs cardanoTx) $
 -- This function has an ambiguous type, so it is probably necessary to
 -- type-apply it to the type of your validator.
 spOutGetDatum :: (Pl.FromData (Pl.DatumType a), MonadBlockChain m) => SpendableOut -> m (Pl.DatumType a)
-spOutGetDatum (_, txOut) = do
-  Just (Pl.Datum datum) <- datumFromTxOut txOut
-  case Pl.fromBuiltinData datum of
-    Just d -> return d
-    Nothing -> fail "could not parse datum to correct type"
+spOutGetDatum spOut = do
+  mDatum <- datumFromSpOut spOut
+  case mDatum of
+    Nothing -> fail "there is no datum on this output"
+    Just datum -> parseDatum datum
+  where
+    parseDatum (Pl.Datum builtinData) =
+      case Pl.fromBuiltinData builtinData of
+        Just d -> return d
+        Nothing -> fail "could not parse datum to correct type"
+
+datumFromSpOut :: MonadBlockChain m => SpendableOut -> m (Maybe Pl.Datum)
+datumFromSpOut (_, cito) = datumFromTxOut cito
```

Comment:
I would advocate for a more verbose variable name. Unless `cito` in a common naming convention, it seems a bit puzzling when you don't have `ChainIndexTxOut` in mind.